### ISSUE TITLE
Store normalisation method explicitly in `DataProcessor`

### ISSUE THREAD

**Issue by tom-andersson** [tom-andersson/deepsensor#27]
Currently, the method used for normalising `xarray` or `pandas` data with a `DataProcessor` object is stored in the `.norm_params` attr as follows:
```
> print(data_processor)
DataProcessor with normalisation params:
{'coords': {'time': {'name': 'time'},
            'x1': {'map': [-3500000.0, 3500000.0], 'name': 'y'},
            'x2': {'map': [-3500000.0, 3500000.0], 'name': 'x'}},
 'mask': {'max': 1.0, 'min': 0.0},
 'surface': {'max': 4047.721923828125, 'min': 0.0},
 't2m': {'mean': 0.05024006962776184, 'std': 3.8340163230895996},
 'tas': {'mean': -16.261720883718514, 'std': 14.695655954303945}}
```
I.e. the keys are `mean`/`std` or `min`/`max`.

It would be better to use a format like `method="mean_std", param1=<mean>, param2=<std>`. Then, the `method` string can be extracted explicitly and `param1` and `param2` used as appropriate for the normalisation/unnormalisation in `DataProcessor`.

**Comment by patel-zeel** [comment]
Hi,@tom-andersson. Would it be helpful to add a log/power normalization option for some datasets that have heavy-tailed log normal distribution? For example, our PM2.5 dataset distribution is as follows in the raw form:
![image](https://github.com/tom-andersson/deepsensor/assets/59758528/7896af7d-8d29-4f23-a54d-b25709f97c30)

Once applied log transform or [PowerTransfrom](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.PowerTransformer.html), it becomes more like a normal distribution.
![image](https://github.com/tom-andersson/deepsensor/assets/59758528/a3a3c46e-56be-4d6f-b89f-e18e9da9905a)

However, there is a trick here. Once the model predicts `mean_ds, std_ds`, we need to transform this multi-variate normal to a multi-variate log normal distribution using the "change of variables" trick.

In theory, it looks good but one needs to test on a toy problem if such a transformation improves the performance of the model.

**Comment by tom-andersson** [comment]
Hey @patel-zeel, this sounds like a nice addition. Would you like to give this a go with a PR once I've addressed the issue above? We don't yet have a `CONTRIBUTIONS.md` but see some comments on submitting a PR [here](https://github.com/tom-andersson/deepsensor/pull/35#issuecomment-1673237937).

**Comment by tom-andersson** [comment]
Okay, the feature mentioned in OP is implemented in https://github.com/tom-andersson/deepsensor/commit/a6c8d0a882571970e0cb4c19889006ae763c4d91. Note that this is a breaking change for `DataProcessor` config, so any existing stored configs will have to be updated. We now have:
```
> ds = data_processor(ds_raw, method="mean_std")
> print(data_processor)
DataProcessor with normalisation params:
{'air': {'method': 'mean_std',
         'param1': 281.255126953125,
         'param2': 16.320409774780273},
 'coords': {'time': {'name': 'time'},
            'x1': {'map': (15.0, 75.0), 'name': 'lat'},
            'x2': {'map': (200.0, 330.0), 'name': 'lon'}}}
```

Or using `min_max`:
```
> ds = data_processor(ds_raw, method="min_max")
> print(data_processor)
DataProcessor with normalisation params:
{'air': {'method': 'min_max', 'param1': 221.0, 'param2': 317.3999938964844},
 'coords': {'time': {'name': 'time'},
            'x1': {'map': (15.0, 75.0), 'name': 'lat'},
            'x2': {'map': (200.0, 330.0), 'name': 'lon'}}}
```

With this config for the `DataProcessor`, we can unnormalise the variable with ID `'air'` without needing to remember the normalisation method we originally chose:
```
ds_unnorm = data_processor.unnormalise(ds)  # Don't need to specify `method` - already stored explicitly in `norm_params`
xr.testing.assert_allclose(ds_unnorm, ds_raw)  # Passes!
```
